src/storage/service.ts

```diff
@@ -627,9 +657,82 @@ export class StorageService {
   }
 
   /**
-   * Select a random provider from the given list
+   * Select a random provider from the given list with ping validation
+   * @param providers - List of available providers
+   * @param signer - Signer for entropy generation
+   * @param excludedProviderAddresses - List of provider addresses to exclude from selection
+   * @param pdpServer - PDPServer instance for ping validation
+   * @returns A provider that responds to ping
+   * @throws Error if no providers are reachable
    */
   private static async selectRandomProvider (
+    providers: ApprovedProviderInfo[],
+    signer: ethers.Signer,
+    excludedProviderAddresses: string[] = [],
+    enablePingValidation: boolean = false
```

Comment:
both of these arguments are not being provided by the caller, I think you have a logic mixup here, maybe you don't need the `excludedProviderAddresses` here _or_ in `selectProviderWithPing`, just pass down your list to `selectProviderWithPing` and let it return the first in your sorted (or random) list and that's the one to go with; I think that works in both cases?